packages/block-editor/src/store/selectors.js

```diff
@@ -1623,3 +1623,26 @@ export function isNavigationMode( state ) {
 export function didAutomaticChange( state ) {
 	return !! state.automaticChangeStatus;
 }
+
```

Comment:
Hi @retrofox,
Selectors are part of the public API, I would prefer to avoid adding these functions to the public API.
As a temporary solution, you may query the values outside a selector relying on getSettings. But ideally, we would use useColors in this case, I'm not sure if it is possible because it seems we relly on the colors of the parent.

Reply:
cc: @epiqueras Do you think is there a way to use the useColors hook in this case to avoid the need to query colors by slug?

Reply:
Yes. The components returned by the hook have a color property that always contains the color value, even for preset colors.

Reply:
We weren't able to use `useColors` directly here because it would try to read color attributes from attributes of the current block. In this case it is always empty because those values actually come from one of the parent blocks. Or am I misreading the usage?

Reply:
The parent block Navigation has `useColors` implemented but this usage is a bit different

Reply:
The tough point here is how to propagate the colors from the Navigation to its children (NavLink).
Right now, we are pulling the attrs from the children, and it works as expected. It is used not only by colors but other attributes such as font-size, alignment, etc.
The colors are handled by the useColors() hook straight in the parent Navigation block, and it's great. 

The issue is we need always the color in RGB format, and it doesn't happen right now since the custom color will be undefined if the color was picked up from colors palette.

It true that we can pick up the color in RGB through of the `.color` the property, but then we need to store this new (lets say rgbColor) attribute into the Navigation in order to be able to pull it from its children. In fact, [it's working this way now in master](https://github.com/WordPress/gutenberg/pull/19681/commits/0af4f79b5d31130db474412959fbb97c63a9c6a4#diff-cfe334a821ad12721a129f61005669c3R94-R101).

For all of this, we suggest removing this attribute and compute the color in the children. I'm going to move this selector outside of the public API.